### PR TITLE
Add HAProxy documentation

### PR DESCRIPTION
**Pull request by scherepn**
If you’re adding a new scraper, please ensure that you have:

- [x] Tested the scraper on a local copy of DevDocs
- [x] Ensured that the docs are styled similarly to other docs on DevDocs

- [x] Added these files to the <code>public/icons/*your_scraper_name*/</code> directory:
  - [x] `16.png`: a 16×16 pixel icon for the doc
  - [x] `16@2x.png`: a 32×32 pixel icon for the doc
  - [x] `SOURCE`: A text file containing the URL to the page the image can be found on or the URL of the original image itself

### CODE REVIEW

**Comment by simon04** [comment]
Amazing work!

I noticed one little glitch, i.e., the anchor links should probably be removed as they link to a non-existing anchor: link is http://localhost:9292/haproxy~2.3/configuration#3.1-cpu-map, but should be http://localhost:9292/haproxy~2.3/configuration#cpu-map



What do you think?

File: lib/docs/scrapers/haproxy.rb [hunk 1]
```diff
@@ -0,0 +1,62 @@
+module Docs
+  class Haproxy < UrlScraper
+    self.type = 'haproxy'
```
Comment (simon04) [inline lib/docs/scrapers/haproxy.rb:3]: Maybe we should set `self.name = 'HAProxy'` for a correct typesetting?

**Comment by scherepn** [comment]
Ah, good catch.
The source documents have 5 or 6 different anchors added in for everything record and I worked over this scraper a few times changing which anchors were used multiple times. Looks like I missed a few.

Removing obvious links that go nowhere is easy, I'll fix it.

**Review by simon04 — APPROVED** [review]
Thank you!

**Pull request merged**